curp/src/keybased_channel.rs

```diff
@@ -12,55 +13,90 @@ use tokio::task::JoinHandle;
 
 use crate::cmd::ConflictCheck;
 
-/// Keys and Messages combined structure
-#[derive(Hash, PartialEq, Eq)]
-pub(crate) struct KeysMessage<K, M> {
+/// The inner type for Keys and Messages combined structure
+struct KeysMessageInner<K, M> {
     /// The keys
     keys: Vec<K>,
     /// The message
     message: M,
 }
 
+impl<K, M> KeysMessageInner<K, M> {
+    /// Create a new `KeysMessageInner` structure
+    fn new(keys: Vec<K>, message: M) -> Self {
+        Self { keys, message }
+    }
+}
+
+/// Keys and Messages combined structure
+pub(crate) struct KeysMessage<K, M> {
+    /// Inner
+    inner: Arc<KeysMessageInner<K, M>>,
```

Comment:
Is it possible to use Rc instead of Arc?

Reply:
It's transfered via async channels, Arc is required.